core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,138 @@
+ */
+class ExceptionReportingServerErrorHandler implements ServerErrorHandler, ServerListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+
+    private final ServerErrorHandler delegate;
+    private final Duration duration;
+    private final CumulativeCounter counter;
+    private long lastExceptionsCount;
+    private boolean started;
+    @Nullable
+    private Throwable thrownException;
+    @Nullable
+    private ScheduledFuture<?> reportingTaskFuture;
+
+    ExceptionReportingServerErrorHandler(ServerErrorHandler serverErrorHandler, Duration duration) {
+        assert !duration.isNegative() && !duration.isZero();
+        delegate = serverErrorHandler;
+        this.duration = duration;
+        counter = new CumulativeCounter(new Id("armeria.server.unhandledExceptions", Tags.empty(),
+                                               null, null, Type.COUNTER));
+    }
+
+    /**
+     * If it's required to report exceptions, increase the counter and store first thrown exception.
+     */
+    @Nullable
+    @Override
+    public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
+        if (ctx.shouldReportUnhandledExceptions()) {
+            if (thrownException == null) {
+                thrownException = cause;
+            }
+            counter.increment();
+        }
+        return delegate.onServiceException(ctx, cause);
+    }
+
+    @Nullable
+    @Override
+    public AggregatedHttpResponse renderStatus(ServiceConfig config, @Nullable RequestHeaders headers,
+                                               HttpStatus status, @Nullable String description,
+                                               @Nullable Throwable cause) {
+        return delegate.renderStatus(config, headers, status, description, cause);
+    }
+
+    @Override
+    public void serverStarting(Server server) throws Exception {}
+
+    @Override
+    public void serverStarted(Server server) throws Exception {
+        if (started) {
+            return;
+        }
+
+        started = true;
+        reportingTaskFuture = server.config().workerGroup().scheduleAtFixedRate(
+                this::reportException, duration.toMillis(), duration.toMillis(), TimeUnit.MILLISECONDS);
+        server.config().meterRegistry().more().counter(counter.getId().getName(), counter.getId().getTags(),
+                                                       counter, CumulativeCounter::count);
+    }
+
+    @Override
+    public void serverStopping(Server server) throws Exception {
+        if (reportingTaskFuture != null) {
+            reportingTaskFuture.cancel(true);
+            reportingTaskFuture = null;
+        }
+        started = false;
+    }
+
+    @Override
+    public void serverStopped(Server server) throws Exception {}
+
+    private void reportException() {
+        final long exceptionsCount = (long) counter.count() - lastExceptionsCount;
```

Comment:
You can extract `(long) counter.count()` to a local variable, then you don't have to do addition at the end of this method but just assignment.

Reply:
Sure thanks !!